src/hooks/useActiveProject.tsx

```diff
@@ -156,13 +156,19 @@ export const ActiveProjectContextProvider: React.FC<ActiveProjectContextProvider
     const state = calculateState();
 
     useEffect(() => {
-      if (
-        !query.isRefetching &&
-        state.status === 'not-a-patient' &&
-        keepWaitingForPatientId
-      ) {
-        query.refetchAll();
-      }
+      const intervalId = setInterval(() => {
```

Comment:
Don't we have logic elsewhere to check for the subject existence? Maybe it would make sense to add an event emitter. Not sure if we have other cases where this might be needed.

Reply:
I will take this is a follow up task after this PR.